src/Kliva/Models/ActivitySummaryCacheLoader.cs

```diff
@@ -0,0 +1,103 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Runtime.InteropServices.WindowsRuntime;
+using System.Threading.Tasks;
+using Windows.Foundation;
+using Windows.UI.Xaml.Data;
+using DynamicData;
+using Kliva.Services;
+
+namespace Kliva.Models
+{
+    internal class ActivitySummaryCacheLoader
+    {
+        public delegate Task<string> LoadMethod(int page, int pageSize);
+
+        private readonly LoadMethod _loader;
+        private readonly Func<string, Task<List<ActivitySummary>>> _hydrater;
+        private readonly int _pageSize;
+        private readonly ISourceCache<ActivitySummary, long> _targetCache;
+        private readonly string _backingStoreCacheName;
+        private int _page;
+        private bool _suppressNextBackingCacheGet = false;
+
+        public ActivitySummaryCacheLoader(
+            LoadMethod loader,
+            Func<string, Task<List<ActivitySummary>>> hydrater,
+            int pageSize, ISourceCache<ActivitySummary, long> targetCache,
+            string backingStoreCacheName)
+        {
+            if (loader == null) throw new ArgumentNullException(nameof(loader));
+            if (hydrater == null) throw new ArgumentNullException(nameof(hydrater));
+            if (targetCache == null) throw new ArgumentNullException(nameof(targetCache));
+            if (backingStoreCacheName == null) throw new ArgumentNullException(nameof(backingStoreCacheName));
+
+            _loader = loader;
+            _hydrater = hydrater;
+            _pageSize = pageSize;            
+            _targetCache = targetCache;
+            _backingStoreCacheName = backingStoreCacheName;
+        }
+
+        public IAsyncOperation<LoadMoreItemsResult> LoadMoreItems(uint count)
+        {
+            return AsyncInfo.Run(_ => LoadMoreItemsAsync());
+        }
+
+        public async void Reset()
+        {
+            _page = 0;
+            _targetCache.Clear();
+            _suppressNextBackingCacheGet = true;
+            await LoadMoreItemsAsync();
+        }
+
+        public bool IsLoading { get; private set; }        
+
+        private async Task<LoadMoreItemsResult> LoadMoreItemsAsync()
+        {
```

Comment:
Also, not sure why we need to trigger loadmoreitems in the constructor fo the ActivitySummaryService class?
In the ViewModel we create the class and do a .Bind to have the observables being hooked up to the listviews.
I would rather try to do the binding as fast as possible and after that trigger the first data load?

Makes sense?